quandl/model/data_mixin.py

```diff
@@ -35,3 +37,10 @@ def to_numpy(self):
 
     def to_csv(self):
         return self.to_pandas().to_csv()
+
+    def validate_col_index(self, df, keep_column_indexes):
+        num_columns = len(df.columns)
+        for col_index in keep_column_indexes:
+            if col_index > num_columns or col_index < 1:
+                raise ColumnNotFound('Requested column index %s does not exist!'
```

Comment:
from carrier: "It's fine as is. I would just remove the !" I'll remove the exclamation mark